src/main/java/seedu/address/model/internship/Role.java

```diff
@@ -18,7 +18,7 @@ public class Role implements Comparable<Role> {
      */
     public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
 
-    private final String role;
+    protected final String role;
```

Comment:
whats the reason for changing this to protected?

Reply:
companyName is also currently public to allow NameContainsKeywordsPredicate to access the field in the test() method.
Following the same idea, role is needed for RoleContainsKeywordsPredicate. 
However, perhaps both could be modified to let the role class handle the check instead.

Reply:
I've changed this to private.